text/0000-circleci-orb.md

```diff
@@ -0,0 +1,124 @@
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- We must support the Orb, which could create additional release burden.
+- We are encouraging a pattern that is sub-optimal in terms of performance (i.e. running pack instead of lifecycle directly)
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Document how to use buildpacks on CircleCI without an Orb
+- Integrate with a different CI/CD platform like Jenkins
+
+## Run lifecycle instead of pack
+
+Instead of a job that runs `pack build`, we could implement a job that runs each of the lifecycle steps in a container. This has the advantage of decoupling from pack and not requiring the Docker daemon (like the [Tekton Template](https://github.com/tektoncd/catalog/blob/master/buildpacks/buildpacks-v3.yaml)). However, there are also some disadvantages:
+
+* Running lifecycle alone would not support download buildpacks as with `--buildpack` option in pack (you'd have to do it manually)
+* Running lifecycle along would not support `project.toml` when it is shipped.
+* There is no mechanism to `create-builder` or `create-package` without pack.
+* Some of the CircleCI features that would make this approach advantagous are paid-only like `docker_layer_caching`
+* CircleCI will not cache the stack images
+
+An example of what this Orb might look like is the following:
+
+```yaml
+jobs:
+  build:
+    working_directory: /workspace
+    docker:
+    - image: << parameters.builder >>
+    parameters:
+      image-name:
+        type: string
+      builder:
+        type: string
+    steps:
+      - checkout
+      - run: mkdir /tmp/cache
+      - restore_cache:
+          key: cnb-cache-<< parameters.image-name >>-<< parameters.builder >>-{{ arch }}
+      - run:
+          command: /cnb/lifecycle/detector
+          name: Detect
+      - run:
+          command: /cnb/lifecycle/restorer -path /tmp/cache
+          name: Restore
+      - run:
+          command: /cnb/lifecycle/analyzer << parameters.image-name >>
+          name: Analyze
+      - run:
+          command: /cnb/lifecycle/builder
+          name: Build
+      - run:
+          command: /cnb/lifecycle/exporter << parameters.image-name >>
+          name: Export
+      - run:
+          command: /cnb/lifecycle/cacher -path /tmp/cache
+          name: Cache
+      - save_cache:
+          key: cnb-cache-<< parameters.image-name >>-<< parameters.builder >>-{{ arch }}-{{ epoch }}
+          paths:
+            - /tmp/cache
+```
+
+In the future, we should implement an orb/job that runs lifecycle independently from pack. Ideally if we have a `/lifecycle/prepare` or similar phase that handles `project.toml` and setup/download of buildpacks. But in order to support the full feature set of pack we will start with the proposal in this RFC.
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [circleci/docker](https://circleci.com/orbs/registry/orb/circleci/docker)
+
+# Unresolved Questions
```

Comment:
@jromero brought up a good question in one of the WG meetings that we should make note of, what is the criteria for platforms we're going to support especially for 3rd party integrations like Tekton, CircleCI, Jenkins, etc?